app/src/main/java/com/infomaniak/mail/ui/main/menu/MoveFragment.kt

```diff
@@ -110,51 +132,26 @@ class MoveFragment : MenuFoldersFragment() {
         )
     }
 
-    private fun setupFolderAdapters() {
-        moveViewModel.getFolderIdAndCustomFolders().observe(viewLifecycleOwner) { (folderId, customFolders) ->
-
-            currentFolderId = folderId
-
-            val defaultFoldersWithoutDraft = mainViewModel.currentDefaultFoldersLive.value!!.let { folders ->
-                folders.filterNot { it.role == FolderRole.DRAFT }
-            }
-
-            defaultFoldersAdapter.setFolders(defaultFoldersWithoutDraft, folderId)
-            customFoldersAdapter.setFolders(customFolders, folderId)
-            setSearchBarUi(allFolders = defaultFoldersWithoutDraft + customFolders)
-        }
-    }
-
-    private fun observeNewFolderCreation() = with(mainViewModel) {
-        newFolderResultTrigger.observe(viewLifecycleOwner) { inputDialog.resetLoadingAndDismiss() }
-        isMovedToNewFolder.observe(viewLifecycleOwner) { isFolderCreated ->
-            if (isFolderCreated) findNavController().popBackStack()
-        }
+    private fun observeFolders() {
+        moveViewModel.getFolders().observe(viewLifecycleOwner, ::setSearchBarUi)
     }
 
-    override fun onFolderSelected(folderId: String): Unit = with(navigationArgs) {
-        mainViewModel.moveThreadsOrMessageTo(folderId, threadsUids.toList(), messageUid)
-        findNavController().popBackStack()
-    }
-
-    override fun onFolderCollapse(folderId: String, shouldCollapse: Boolean) = Unit
-
     private fun setSearchBarUi(allFolders: List<Folder>) = with(binding) {
-        searchResultsList.adapter = searchResultsAdapter
 
         searchInputLayout.setOnClearTextClickListener { trackMoveSearchEvent(SEARCH_DELETE_NAME) }
 
         searchTextInput.apply {
-            toggleFolderListsVisibility(!text.isNullOrBlank())
+            toggleFolderListsVisibility(text?.toString(), allFolders)
+
             doOnTextChanged { newQuery, _, _, _ ->
-                toggleFolderListsVisibility(!newQuery.isNullOrBlank())
+                toggleFolderListsVisibility(newQuery?.toString(), allFolders)
                 if (newQuery?.isNotBlank() == true) {
                     moveViewModel.filterFolders(newQuery.toString(), allFolders, shouldDebounce = true)
                 }
 
-                if (!hasAlreadyTrackedFolderSearch) {
+                if (!hasAlreadyTrackedSearch) {
                     trackMoveSearchEvent("executeSearch")
-                    hasAlreadyTrackedFolderSearch = true
+                    hasAlreadyTrackedSearch = true
```

Comment:
Most of the `setSearchBarUi()` code should be executed as fast as possible a single time inside the `onViewCreated()`, no need to wait for a liveData nor update as the liveData updates

Reply:
Most of the `setSearchBarUi()` code needs this liveData value.
What do you want to do exactly ?

Reply:
It's already triggered a single time. It won't update because the liveData can't emit updates.

Reply:
We still wait for no reason to execute most of this code which also has the effect of not splitting concerns as well as we could and it impacts readability and understanding. We should only execute the part that actually needs to wait on the liveData inside the observer I think

Reply:
Ah if most of the code depends on the folders and it's only called once anyway, then there's no problem. I didn't look through it precisely enough